include/envoy/http/header_map.h

```diff
@@ -23,12 +23,6 @@ class LowerCaseString {
   LowerCaseString(LowerCaseString&& rhs) : string_(std::move(rhs.string_)) {}
   LowerCaseString(const LowerCaseString& rhs) : string_(rhs.string_) {}
   explicit LowerCaseString(const std::string& new_string) : string_(new_string) { lower(); }
-  explicit LowerCaseString(std::string&& new_string, bool convert = true)
```

Comment:
I think this used to be a fast path in the h2 codec, but from a quick look of the code this must have been replaced at some point with `addViaMove` which assumes it's already lower case. (We should probably have some type of ASSERT inside HeaderMapImpl for this case which I don't see...)

Reply:
Ooooh, that's a good thing to be aware of, because I'm just now writing (non-upstream) code using add-via-move which is not lower casing.  I'll add the assert in a follow-up since I don't think it logically belongs here.

Reply:
On second thought, I think we're good here.  The only case we don't lower() is when we're copying an existing LowerCaseString, which was lower()ed in the first place.

Reply:
Right, that's true. `addViaMove()` is a special function not in the interface that is used by the codecs to insert data directly. IMO it would still be good to have an ASSERT that the key is lower case for this function, but that's more just an observation and a defense in depth thing.